### PR TITLE
[Merged by Bors] - feat: Composition.blocksFun_le

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: Mathlib/Algebra/Order/BigOperators/Group/List.lean [hunk 1]
```diff
@@ -174,5 +174,17 @@ variable [CanonicallyOrderedCommMonoid M] {l : List M}
     exact le_self_mul
   · simp [take_of_length_le h, take_of_length_le (le_trans h (Nat.le_succ _))]
 
+@[to_additive] theorem le_prod_of_mem
```
Comment (b-mehta) [inline Mathlib/Algebra/Order/BigOperators/Group/List.lean:177]: This looks like an alternate statement of `single_le_prod`, and as such I think they should both have docstrings referencing the other. It also looks like this name is inconsistent with that one, although I personally prefer this name

File: Mathlib/Combinatorics/Enumerative/Composition.lean [hunk 2]
```diff
@@ -168,6 +168,12 @@ theorem blocks_pos' (i : ℕ) (h : i < c.length) : 0 < c.blocks[i] :=
 theorem one_le_blocksFun (i : Fin c.length) : 1 ≤ c.blocksFun i :=
   c.one_le_blocks (c.blocksFun_mem_blocks i)
 
+theorem blocksFun_le {n} (c : Composition n) (i : Fin c.length) :
+    c.blocksFun i ≤ n := by
+  have := c.blocks_sum
+  have := List.le_sum_of_mem (c.blocksFun_mem_blocks i)
+  simp_all
```
Comment (b-mehta) [inline Mathlib/Combinatorics/Enumerative/Composition.lean:175]: Out of curiosity, does `simp_all [c.blocks_sum, List.le_sum_of_mem (c.blocksFun_mem_blocks i)]` work here?

File: Mathlib/Algebra/Order/BigOperators/Group/List.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/Algebra/Order/BigOperators/Group/List.lean:177]: They have quite different hypotheses. I'll add cross-references in doc-strings.

File: Mathlib/Combinatorics/Enumerative/Composition.lean (hunk 2 above)
Comment (semorrison) [reply Mathlib/Combinatorics/Enumerative/Composition.lean:175]: No.

**Comment by b-mehta** [comment]
Thanks!

bors merge

**Pull request closed without merging**